packages/@aws-cdk/toolkit-lib/lib/api/drift/drift.ts

```diff
@@ -0,0 +1,87 @@
+import { format } from 'util';
+import type { DescribeStackDriftDetectionStatusCommandOutput, DescribeStackResourceDriftsCommandOutput } from '@aws-sdk/client-cloudformation';
+import { ToolkitError } from '../../toolkit/toolkit-error';
+import type { ICloudFormationClient } from '../aws-auth/private';
+import type { IoHelper } from '../io/private';
+import { IO } from '../io/private';
+
+/**
+ * Detect drift for a CloudFormation stack and wait for the detection to complete
+ *
+ * @param cfn - a CloudFormation client
+ * @param ioHelper - helper for IO operations
+ * @param stackName - the name of the stack to check for drift
+ * @returns the CloudFormation description of the drift detection results
+ */
+export async function detectStackDrift(
+  cfn: ICloudFormationClient,
+  ioHelper: IoHelper,
+  stackName: string,
+): Promise<DescribeStackResourceDriftsCommandOutput> {
+  await ioHelper.notify(IO.CDK_TOOLKIT_I4506.msg(format('Starting drift detection for stack %s...', stackName)));
+
+  // Start drift detection
+  const driftDetection = await cfn.detectStackDrift({
+    StackName: stackName,
+  });
+
+  await ioHelper.notify(IO.CDK_TOOLKIT_I4506.msg(format('Detecting drift for stack %s...', stackName)));
```

Comment:
Is this good UX? Seems like we are just printing a non actionable message twice. And as soon as we are hitting `waitForDriftDetection` we are sending a third message that's also just "Waiting for"

I'd suggest to keep the first message but use the wording if the second.

Reply:
Actually the second message be turned into a debug containing the StackDriftDetectionId